roles/ceph-rgw/tasks/rgw_create_pools.yml

```
@@ -73,12 +73,39 @@
     - item.value.type is not defined or item.value.type == 'replicated'
     - item.value.rule_name | default(ceph_osd_pool_default_crush_rule_name)
 
+- name: get pool details
+  command: "{{ container_exec_cmd }} ceph --connect-timeout=10 --cluster={{ cluster }} --format=json osd pool ls detail"
+  register: result
+  retries: 60
+  delay: 3
+  until: result is succeeded
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  changed_when: false
+  check_mode: false
+
+- name: set rgw_osd_pool_ls_detail fact
+  set_fact:
+    rgw_osd_pool_ls_detail: "{{ result.stdout | from_json }}"
+
 - name: set the rgw_create_pools pools application to rgw
   command: "{{ container_exec_cmd }} ceph --connect-timeout 10 --cluster {{ cluster }} osd pool application enable {{ item.key }} rgw"
   register: result
   retries: 60
   delay: 3
   until: result is succeeded
-  changed_when: false
   loop: "{{ rgw_create_pools | dict2items }}"
   delegate_to: "{{ groups[mon_group_name][0] }}"
+  when: >
+    'rgw' not in (rgw_osd_pool_ls_detail | selectattr('pool_name', 'eq', item.key) | first).application_metadata
```

```suggestion
    'rgw' not in (rgw_osd_pool_ls_detail | selectattr('pool_name', 'match', '^'+item.key+'$') | first).application_metadata
```

See https://github.com/ceph/ceph-ansible/commit/487dcdc3f05a0de6e486e6754ac42576440eec51

---

Sorry for the delay.

I'm not a huge fan of this workaround, as it will lead to weird behavior if pool names contain `$` or `*` or `.` etc. I wonder if it wouldn't be better to add Jinja2 >= 2.8 to `requirements.txt` and let EL7 users install it using `pip` if necessary.

Nevertheless, I've added a commit to implement the suggested workaround.

---

@dsavineau what do you think about a potential bump to Jinja2 >= 2.8?
couldn't it be a problem to bump like this in the middle of a ceph-ansible release?